packages/near-fast-auth-signer/src/utils/multi-chain/multiChain.ts

```diff
@@ -0,0 +1,166 @@
+
+      txid = (await evm.handleTransaction(
+        req.transaction,
+        req.account,
+        req.transaction.derivedPath,
+      )).hash;
+    } else if (req.chainConfig.type === 'BTC') {
+      const btc = new Bitcoin({ ...req.chainConfig, relayerUrl: req.fastAuthRelayerUrl });
+
+      txid = await btc.handleTransaction(
+        { ...req.transaction, value: parseFloat(req.transaction.value) },
+        req.account,
+        req.transaction.derivedPath,
+      );
+    } else {
+      throw new Error('Unsupported chain type');
+    }
+
+    return {
+      transactionHash: txid,
+      success:         true,
+    };
+  } catch (e) {
+    return {
+      success:      false,
+      errorMessage: e.message
+    };
+  }
+};
+
+export const getDerivedAddress = async (
+  signerId: string,
+  path: string,
+  chainConfig: ChainConfig,
+  account: Account,
+  contract: ChainSignatureContracts,
+  relayerUrl: string
+) => {
+  let derivedAddress: string;
+
+  switch (chainConfig.type) {
+    case 'EVM':
+      derivedAddress = await EVM.deriveAddress(signerId, path, account, contract, relayerUrl);
+      break;
+    case 'BTC':
+      derivedAddress = (await Bitcoin.deriveAddress(
+        signerId,
+        path,
+        chainConfig.networkType === 'testnet' ? bitcoin.networks.testnet : bitcoin.networks.bitcoin,
+        account,
+        contract,
+        relayerUrl
+      )).address;
+      break;
+    default:
+      throw new Error('Unsupported chain config');
+  }
+
+  return derivedAddress;
+};
+
+/**
+ * Calculates the estimated maximum fee for an EVM transaction.
+ *
+ * @param {Object} transaction - The transaction details including the recipient's address, value, and data.
+ * @param {EVMChainConfigWithProviders} chainConfig - The configuration object for the EVM chain, including provider URLs and other relevant settings.
+ * @returns {Promise<bigint>} The estimated maximum transaction fee in wei.
+ */
+export const getEstimatedFeeEVM = async (
+  transaction: {
+  to: string;
+  value?: BigNumberish;
+  data?: string;
+},
+  chainConfig: EVMChainConfigWithProviders,
+  relayerUrl: string
+): Promise<bigint> => {
+  const evm = new EVM({ ...chainConfig, relayerUrl });
+  return (await evm.getFeeProperties(transaction)).maxFee;
+};
+
+/**
+ * Calculates the estimated fee for a Bitcoin transaction in satoshis.
+ *
+ * @param {Object} transaction - The transaction details including the sender's address and the targets.
+ * @param {string} transaction.from - The Bitcoin address from which the transaction is sent.
+ * @param {Array} transaction.targets - An array of objects, each containing the target address and value in satoshis to send.
+ * @param {BTCChainConfigWithProviders} chainConfig - The configuration object for the Bitcoin chain, including network type and providers.
+ * @returns {Promise<number>} The estimated transaction fee in satoshis.
+ */
+export const getEstimatedFeeBTC = async (transaction: {
```

Comment:
Just like `signAndSend` and `getDerivedAddress`, can we also wrap `getEstimatedFeeEVM` and `getEstimatedFeeBTC` into a single function with switch case logic?

If you are concerned about maintaining the gas fee unit, we can also return like:
{ type: 'gwei', totalTransactionFee: NUMBER }

Reply:
That's a good point that me and @esaminu discussed, we can have a discussion on this.

But the main issues with a single method are:

- Complexity: As more chains are added, the function's argument types and body will become increasingly complex.
- Usability: Without TypeScript, it will be challenging for callers to use and understand the function correctly, making documentation and usage harder.

Reply:
> Complexity: As more chains are added, the function's argument types and body will become increasingly complex.
Wouldn't EVM ones will have similar types? even if we have more and more support on different coins, at some point many of them will overlapp. 

> Usability: Without TypeScript, it will be challenging for callers to use and understand the function correctly, making documentation and usage harder.
We could potentially do like `Arg type = type A | type B ....` and apply types separately on calling dedicated functions inside switch statement.

As we discussed, at the end of the day, would be nicer if user just simply calls top level function and get the fee data that they want. But if this is too time consuming, I'm happy to go with what we have

Reply:
Fully agree with calling only top level function. But what we are afraid it's that this functions will get very complex. 

I'm open to set a call and talk about it, maybe a single function it's idd better. 

I will create a ticket for it and we discuss for the module creation